app/logic/Chat/ChatAccount.ts

```diff
@@ -28,8 +28,8 @@ export class ChatAccount extends TCPAccount {
   }
 
   async save(): Promise<void> {
-    await super.save();
     await this.storage?.saveAccount(this);
+    await super.save();
```

Comment:
I understand that this change was necessary to make it work, and that it works.

But this makes assumptions that shouldn't be there. This relies on subtle ordering, due to completely unrelated code which doesn't appear here. This is very likely to regress.

It also makes assumptions about the mail account being the primary account and the chat account being the secondary one. We should avoid such assumptions, if possible. And particularly, such assumptions must never be implicit, but either clearly obvious from the code, or documented on both sides of the assumption/link.

Can you find a fix that does not rely on this ordering, please?

Reply:
There will always be an ordering, since the database won't allow a dependent account to be saved before its main account, so the only solution I could come up with was to make the ordering more explicit.

(Note that an ordering also applies to `deleteIt()` in that the `super.deleteIt();` must appear first.)